content/articles/create-a-weather-app-in-node.js-using-openweathermap-api/index.md

```diff
@@ -0,0 +1,560 @@
+                let weatherTemp = `${weather.main.temp}`,
+                    weatherPressure = `${weather.main.pressure}`,
+                    /* We shall fetch the weather icon and its size using the icon data*/
+                    weatherIcon = `http://openweathermap.org/img/wn/${weather.weather[0].icon}@2x.png`,
+                    weatherDescription = `${weather.weather[0].description}`,
+                    humidity = `${weather.main.humidity}`,
+                    clouds = `${weather.clouds.all}`,
+                    visibility = `${weather.visibility}`,
+                    main = `${weather.weather[0].main}`,
+                    weatherFahrenheit;
+                weatherFahrenheit = ((weatherTemp * 9 / 5) + 32);
+
+                // We shall also round off the value of the degrees fahrenheit calculated into two decimal places
+                function roundToTwo(num) {
+                    return +(Math.round(num + "e+2") + "e-2");
+                }
+                weatherFahrenheit = roundToTwo(weatherFahrenheit);
+```
+
+- The stored values shall now be rendered unto the webpage to generate a static webpage which shall be displayed.
+
+```javascript
+                // We shall now render the data to our page (index.ejs) before displaying it out
+                res.render('index', { weather: weather, place: place, temp: weatherTemp, pressure: weatherPressure, icon: weatherIcon, description: weatherDescription, timezone: weatherTimezone, humidity: humidity, fahrenheit: weatherFahrenheit, clouds: clouds, visibility: visibility, main: main, error: null });
+            }
+        }
+    });
+});
+```
+
+> **Note:** You can fetch and display much data as you will need in your project as needed in the json received.
+
+- Set up our port configurations
+  We shall utilize port 5000 for our project which can be accessed on `localhost:5000`.
+
+```javascript
+// We shall set up our port configurations
+app.listen(5000, function() {
+    console.log('Weather app listening on port 5000!');
+});
+```
+
+You can just copy the code and paste it into your file.
+
+#### Setup our views
+
+Let's now set up our dynamic web pages in the application. Remember, we are using **EJS** as our template engine.
+Inside the 'index.ejs' file found in the 'views' folder, we shall create a webpage template that shall be converted into a static webpage during display once the values are fetched. To learn more on template engines, visit '[Getting Started with EJS Templating Engine](https://www.section.io/engineering-education/nodejs-ejs/ "Getting Started with EJS Templating Engine Section.io Blog")' blog.
+
+> **Note:** When this shall render the static pages, they shall be served from the 'public' folder, it acts as the root directory. This is because of this line of code, `app.use(express.static('public'));` in the 'server.js' file. Therefore, it shall obtain its assets from the 'public' folder. This will include the 'css' or even image files.
```

Comment:
You use `shall` a lot. While it is grammatically correct, it has a formal obligatory tone to it that can be intimidating, try switching it or mixing it with will from time to time.

Reply:
Done! :heavy_check_mark: